xds/internal/balancer/clusterimpl/clusterimpl.go

```diff
@@ -302,51 +331,49 @@ func (b *clusterImplBalancer) UpdateSubConnState(sc balancer.SubConn, s balancer
 }
 
 func (b *clusterImplBalancer) Close() {
-	b.serializer.TrySchedule(func(_ context.Context) {
-		b.child.Close()
-		b.childState = balancer.State{}
+	b.child.Close()
+	b.childState = balancer.State{}
 
-		if b.cancelLoadReport != nil {
-			b.cancelLoadReport()
-			b.cancelLoadReport = nil
-		}
-		b.logger.Infof("Shutdown")
-	})
-	b.serializerCancel()
-	<-b.serializer.Done()
+	if b.cancelLoadReport != nil {
+		b.cancelLoadReport()
+		b.cancelLoadReport = nil
+	}
+	b.logger.Infof("Shutdown")
 }
 
 func (b *clusterImplBalancer) ExitIdle() {
-	b.serializer.TrySchedule(func(context.Context) {
-		b.child.ExitIdle()
-	})
+	b.child.ExitIdle()
 }
 
 // Override methods to accept updates from the child LB.
 
 func (b *clusterImplBalancer) UpdateState(state balancer.State) {
-	b.serializer.TrySchedule(func(context.Context) {
-		b.childState = state
-		b.ClientConn.UpdateState(balancer.State{
-			ConnectivityState: b.childState.ConnectivityState,
-			Picker: b.newPicker(&dropConfigs{
-				drops:           b.drops,
-				requestCounter:  b.requestCounter,
-				requestCountMax: b.requestCountMax,
-			}),
-		})
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	// Inhibit sending a picker update to our parent as part of handling new
+	// state from the child, if we are currently handling an update from our
+	// parent. Update the childState field regardless.
+	b.childState = state
+	if b.inhibitPickerUpdates {
+		return
+	}
+
+	b.ClientConn.UpdateState(balancer.State{
+		ConnectivityState: state.ConnectivityState,
+		Picker:            b.newPickerLocked(),
 	})
```

Comment:
Don't we need a call to `newPickerUpdated` after calling `UpdateState` here?

Reply:
Yes, you're right. Updated the PR.